### PR TITLE
Mass Merge - The Blood System Rework & A metric fuckton of runtime fixes

### PR DESCRIPTION
**Pull request by ArchieBeepBoop**
## About The Pull Request

**Please don't let this stay stale for too long otherwise merge conflicts will happen.**

In preparation for the components system update, the reagent ID refactor, viro changes and other things that will allow us to finally get TGUI Next alongside other new features, this PR mass merges the following PRs, modified to work with hyperstation:

https://github.com/Citadel-Station-13/Citadel-Station-13/pull/9047
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/9688
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10199/files
Partial https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10705/files
https://github.com/Citadel-Station-13/Citadel-Station-13/pull/10851

image

## Why It's Good For The Game

I spent way too much time on this port please god send help. This also fixes all the runtimes we had with gibbing and forensics.

## Changelog
:cl:
Please check the attached PRs. There are way too many changes.

TL;DR Slimes bleed, blood can be converted into universal blood for all species, iron has an overdose and only works if you have blood in the first place, species have different blood color types. In the future, slimes will bleed their primary color.
/:cl:

### CODE REVIEW

**Comment by ArchieBeepBoop** [comment]
image
**Do you like hurting other people?**

**Review by Dahlular — APPROVED** [review]
